src/ext4fs.rs

```diff
@@ -3,63 +3,68 @@
 //
 
 use std::ffi::OsString;
-use std::fs::{File};
+use std::io::{Read, Seek};
+use std::path::Path;
 
+use anyhow::anyhow;
+use anyhow::Context;
 use fuse_mt::*;
 
+trait BlockDevice: Read + Seek {}
+
 pub struct Ext4FS {
     pub target: OsString,
-    pub superblock: Option<ext4::SuperBlock<std::io::BufReader<std::fs::File>>>,
+    pub superblock: ext4::SuperBlock<bootsector::RangeReader<std::io::BufReader<std::fs::File>>>,
 }
 
-impl Ext4FS
-{
-    pub fn new(target: OsString) -> Self {
-        println!("Ext4FS.new()");
+impl Ext4FS {
+    pub fn new(target: OsString) -> anyhow::Result<Self> {
+        let mut block_device = std::io::BufReader::new(std::fs::File::open(&target).unwrap());
+        let partitions =
```

Comment:
The problem with detecting partitions here is that it breaks mounting a bare partition on a device directly:

```
$ sfdisk -l /dev/sda | tail -n 2
Device     Start        End    Sectors   Size Type
/dev/sda1  65535 1757910839 1757845305 838.2G Linux filesystem

$ cargo run /dev/sda1 ./mnt
    Finished dev [unoptimized + debuginfo] target(s) in 0.06s
     Running `target/debug/fuse-ext4 /dev/sda1 ./mnt`
Error: searching for partitions

Caused by:
    entity not found

```

Ideally it'd be nice to be able to mount real device partitions directly like this.